t/re-split.t

```diff
@@ -0,0 +1,707 @@
+                return
+            end
+
+            for i = 1, #my_table do
+                ngx.say(res[i])
+            end
+        }
+    }
+--- request
+GET /re
+--- response_body
+a
+,
+b
+,
+c,d,e
+--- error_log eval
+qr/\[TRACE   \d+/
+--- no_error_log
+[error]
+
+
+
+=== TEST 16: split matches, ctx arg
+--- http_config eval: $::HttpConfig
+--- config
+    location = /re {
+        content_by_lua_block {
+            local ngx_re = require "ngx.re"
+
+            local res, err = ngx_re.split("a,b,c,d,e", ",", nil, { pos = 5 })
+            if err then
+                ngx.log(ngx.ERR, "failed: ", err)
+                return
+            end
+
+            for i = 1, #res do
+                ngx.say(res[i])
+            end
+        }
+    }
+--- request
+GET /re
+--- response_body
+c
+d
+e
+--- error_log eval
+qr/\[TRACE   \d+/
+--- no_error_log
+[error]
+
+
+
+=== TEST 17: split matches, trailing subjects
+--- http_config eval: $::HttpConfig
+--- config
+    location /re {
+        content_by_lua_block {
+            local ngx_re = require "ngx.re"
+
+            local res, err = ngx_re.split(",a,b,c,d,", ",")
+            if err then
+                ngx.log(ngx.ERR, "failed: ", err)
+                return
+            end
+
+            for i = 1, #res do
+                if res[i] == "" then
+                    ngx.say("_blank_")
+                else
+                    ngx.say(res[i])
+                end
+            end
+        }
+    }
+--- request
+GET /re
+--- response_body
+_blank_
+a
+b
+c
+d
+_blank_
+--- error_log eval
+qr/\[TRACE   \d+/
+--- no_error_log
+[error]
+attempt to get length of local 'regex' (a number value)
+
+
+
+=== TEST 18: split matches, real use-case
+--- http_config eval: $::HttpConfig
+--- config
+    location /re {
+        content_by_lua_block {
+            local ngx_re = require "ngx.re"
+
+            local res, err = ngx_re.split("abcd,erfg,ghij;hello world;aaa", ",|;")
+            if err then
+                ngx.log(ngx.ERR, "failed: ", err)
+                return
+            end
+
+            for i = 1, #res do
+                ngx.say(res[i])
+            end
+        }
+    }
+--- request
+GET /re
+--- response_body
+abcd
+erfg
+ghij
+hello world
+aaa
+--- error_log eval
+qr/\[TRACE   \d+/
+--- no_error_log
+[error]
+
+
+
+=== TEST 19: split no matches
+--- http_config eval: $::HttpConfig
+--- config
+    location /re {
+        content_by_lua_block {
+            local ngx_re = require "ngx.re"
+
+            local res, err = ngx_re.split("abcd", ",")
+            if err then
+                ngx.log(ngx.ERR, "failed: ", err)
+                return
+            end
+
+            for i = 1, #res do
+                ngx.say(res[i])
+            end
+        }
+    }
+--- request
+GET /re
+--- response_body
+abcd
+--- error_log eval
+qr/\[TRACE   \d+/
+--- no_error_log
+[error]
+
+
+
+=== TEST 20: subject is not a string type
+--- http_config eval: $::HttpConfig
```

Comment:
Valgrind finds a memory invalid read issue while running this test case on my side (Linux x86_64):

```
t/re-split.t .. # I found ONLY: maybe you're debugging?
TEST 20: subject is not a string type
==44835== Invalid read of size 16
==44835==    at 0x4024080: ???
==44835==    by 0x905CE9F: ???
==44835==    by 0xFFEFF64AF: ???
==44835==  Address 0x905cea0 is 16 bytes inside a block of size 27 alloc'd
==44835==    at 0x4C28C50: malloc (in /usr/lib64/valgrind/vgpreload_memcheck-amd64-linux.so)
==44835==    by 0x4C2AB1E: realloc (in /usr/lib64/valgrind/vgpreload_memcheck-amd64-linux.so)
==44835==    by 0x54690BF: lj_mem_realloc (lj_gc.c:811)
==44835==    by 0x546AE40: lj_str_new (lj_str.c:188)
==44835==    by 0x5473ABD: lj_strfmt_int (lj_strfmt.c:356)
==44835==    by 0x5465AFA: lj_ff_tostring (in /opt/luajit21sysm/lib/libluajit-5.1.so.2.1.0)
==44835==    by 0x4D5B94: ngx_http_lua_run_thread (ngx_http_lua_util.c:1005)
==44835==    by 0x4D96F0: ngx_http_lua_content_by_chunk (ngx_http_lua_contentby.c:120)
==44835==    by 0x4D98C5: ngx_http_lua_content_handler_inline (ngx_http_lua_contentby.c:310)
==44835==    by 0x4D92AF: ngx_http_lua_content_handler (ngx_http_lua_contentby.c:222)
==44835==    by 0x45581D: ngx_http_core_content_phase (ngx_http_core_module.c:1379)
==44835==    by 0x4504E3: ngx_http_core_run_phases (ngx_http_core_module.c:856)
==44835==    by 0x4505F1: ngx_http_handler (ngx_http_core_module.c:839)
==44835==    by 0x45BC58: ngx_http_process_request (ngx_http_request.c:1916)
==44835==    by 0x45C034: ngx_http_process_request_headers (ngx_http_request.c:1348)
==44835==    by 0x45C349: ngx_http_process_request_line (ngx_http_request.c:1027)
==44835==    by 0x45C783: ngx_http_wait_request_handler (ngx_http_request.c:504)
==44835==    by 0x444A18: ngx_epoll_process_events (ngx_epoll_module.c:900)
==44835==    by 0x43B189: ngx_process_events_and_timers (ngx_event.c:242)
==44835==    by 0x443A6E: ngx_single_process_cycle (ngx_process_cycle.c:309)
==44835==    by 0x41D0B0: main (nginx.c:364)
==44835==
```